controllers/provisioners/config.go

```diff
@@ -151,14 +179,31 @@ func isConflict(defaultVal, resourceVal interface{}) bool {
 	return false
 }
 
-func (c *ProvisionerConfiguration) setSharedFields(obj map[string]interface{}) error {
+func getMatchingConditionals(ig *v1alpha1.InstanceGroup, conditionals []Conditional) []Conditional {
+	var applicableConditionals = make([]Conditional, 0)
+	for _, conditional := range conditionals {
+		annotationKeyValue := strings.Split(conditional.Annotation, "=")
+		if ig.Annotations[annotationKeyValue[0]] == annotationKeyValue[1] {
```

Comment:
You can actually use providers/kubernetes/utils.go `HasAnnotation(ig.Annotations, annotationKeyValue[0], annotationKeyValue[1])`, might also want to pre validate the input `conditional.Annotation` is not empty and contains '=' sign to avoid crashes in case of bad input